tensorflow/core/kernels/mkl_conv_grad_filter_ops.cc

```diff
@@ -485,13 +485,27 @@ class MklConvCustomBackpropFilterOp
       diff_filter_mkl_shape.SetMklTensor(false);
 
       if (is_conv2d) {
-        // Conv2D: output_dims_mkl_order is in OIHW format.
-        TensorShape diff_filter_tf_shape({bwd_output_dims[MklDnnDims::Dim_H],
-                                          bwd_output_dims[MklDnnDims::Dim_W],
-                                          bwd_output_dims[MklDnnDims::Dim_I],
-                                          bwd_output_dims[MklDnnDims::Dim_O]});
-        AllocateOutputSetMklShape(context, 0, &diff_filter_tensor,
-                                  diff_filter_tf_shape, diff_filter_mkl_shape);
+        if (!is_depthwise) {
+          // Conv2D: output_dims_mkl_order is in OIHW format.
+          TensorShape diff_filter_tf_shape(
+              {bwd_output_dims[MklDnnDims::Dim_H],
+               bwd_output_dims[MklDnnDims::Dim_W],
+               bwd_output_dims[MklDnnDims::Dim_I],
+               bwd_output_dims[MklDnnDims::Dim_O]});
+          AllocateOutputSetMklShape(context, 0, &diff_filter_tensor,
+                                    diff_filter_tf_shape,
+                                    diff_filter_mkl_shape);
+        } else {
+          // Depthwise Conv2d: bwd_output_dims is GOIHW format
```

Comment:
It's not clear to me how `GOIHW` maps to `HWGO` below. Could you please explain more in the comment? Thank you!

Reply:
| TensorFlow       | MKLDNN
--------------------------------------------------------------------
filter_out_depth | depth_multiplier | depth_multiplier * group_count
filter_in_depth  | in_depth         | in_depth / group_count
For depthwise convolution, we have group_count == in_depth.

So here G = original I, and I = 1.
And the GOIHW is mkldnn format, here is try to extract the TF format, TF format is HWIO, as G = original I, so here is HWGO.

Reply:
Understood. Thank you! Could you please also add the explanation as comment in the code?

Reply:
Thank you very much!